Sources/ApolloCodegenLib/Frontend/JavaScript/src/compiler/index.ts

```diff
@@ -127,10 +138,9 @@ export function compileToIR(
     );
 
     const source = print(operationDefinition);
-    const rootType = getOperationRootType(
-      schema,
-      operationDefinition
-    ) as GraphQLObjectType;
+    const rootType = schema.getRootType(operationType) as GraphQLObjectType;
+
+    referencedTypes.add(getNamedType(rootType));
```

Comment:
Do I understand this change correctly that this is where `Query` is being added?

Reply:
That's correct!